src/anndata/_core/views.py

```diff
@@ -305,6 +306,11 @@ def as_view_dask_array(array, view_args):
 
 @as_view.register(pd.DataFrame)
 def as_view_df(df, view_args):
+    if settings.should_remove_unused_categories:
+        for col in df.columns:
+            if isinstance(df[col].dtype, pd.CategoricalDtype):
+                with pd.option_context("mode.chained_assignment", None):
+                    df[col] = df[col].cat.remove_unused_categories()
```

Comment:
I genuinely do not know.  This _seems_ to be unintentional (i.e., missed spots to do this) but the original feature request for the setting seemed only for deduping the `__init__` and/or `obs` (and this is run on `obsm` dataframe views since dedupping happens for `obs` on `__init__`).  I would be in favor of a patch since we are moving to `0.11.0` anyway so can break whatever.